bot/cogs/bot.py

```diff
@@ -131,19 +150,37 @@ def codeblock_stripping(self, msg: str):
 
                 # Strip again to remove the whitespace(s) left before the code
                 # If the msg looked like "Python <code>" before removing Python
-                content = content.strip()
+                # And strips REPL code out of the message if there is any
+                content = self.repl_stripping(content.strip())
                 log.trace(f"Returning message.\n\n{content}\n\n")
                 return content
 
     async def on_message(self, msg: Message):
-        if not msg.author.bot:
-            if msg.channel.id in self.channel_cooldowns:
-                on_cooldown = time.time() - self.channel_cooldowns[msg.channel.id] < 300
-                if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
-                    try:
+        if msg.channel.id in self.channel_cooldowns and not msg.author.bot:
+            on_cooldown = time.time() - self.channel_cooldowns[msg.channel.id] < 300
+            if not on_cooldown or msg.channel.id == DEVTEST_CHANNEL:
+                try:
+                    not_backticks = ["'''", '"""', "´´´", "‘‘‘", "’’’", "′′′", "“““", "”””", "″″″", "〃〃〃"]
+                    python_syntax = any(
+                        msg.content[3:9].lower() == "python",
+                        msg.content[3:5].lower() == "py"
+                    )
+
+                    bad_ticks = msg.content[:3] in not_backticks
+                    if bad_ticks and python_syntax:
+                        howto = (f"Hey {msg.author.mention}!\n"
+                                 "I noticed you were trying to paste code into this channel.\n"
+                                 f"You almost did it but used {not_backticks.index(msg.content[3:])} instead of ```")
+
+                    else:
+                        howto = ""
+                        if bad_ticks:
+                            howto += ("You are using the wrong ticks, use ``` instead of "
+                                      f"{not_backticks.index(msg.content[:3])}\n\n")
+
+                            msg.content = msg.contet.replace(not_backticks.index(msg.content[3:]), "```")
+
                         content = self.codeblock_stripping(msg.content)
```

Comment:
this can return None, but then you feed it into the parse, and parse crashes with a TypeError if you pass in None. Please account for this.

Reply:
@hargoniX Did you address this review comment?

Reply:
yeah, this is addressed on line 186